Add GitHub Actions workflow for build and release

SirDank opened the pull request:
This workflow automates the build and release process for the Knight Launcher project, including fetching dependencies, building the project, and creating a GitHub release.

SirDank committed:
Add GitHub Actions workflow for build and release

This workflow automates the build and release process for the Knight Launcher project, including fetching dependencies, building the project, and creating a GitHub release.

SirDank committed:
fix tag_name

SirDank committed:
Update LauncherApp.java

SirDank committed:
Update build-and-release.yml

SirDank commented:
@lucasluqui It is currently set to only run manually since it also creates the release with the zip attached

lucasluqui commented:
Haven't forgotten about this despite looking like it!

Should be merged before I start developing 2.5 stuff.

lucasluqui committed:
Updated detailed guide URL in README.

lucasluqui committed:
Tag the new release as prerelease

Not fearless enough to send this as a straight up release... even more so considering it will begin serving to current users as soon as it's created. Let's tag it as prerelease so I can test that it works correctly. Will revert later.

SirDank reviewed (commented):
aight cool
